### DISCUSSION TITLE
Application level middleware and hot reloads with Revise

### DISCUSSION THREAD

**Discussion by fypc** [OxygenFramework/Oxygen.jl#230] [Q&A]
Hi everyone,

I am trying to setup an application level middleware while using hot reloads with Revise, adding the `middleware` argument to the `serve` function:

```julia
using Revise
using Oxygen
using MyModule

MyModule.serve(host="0.0.0.0", port=5050, revise=:eager, middleware=[CorsMiddleware])
```
I get the following error: 
```julia
LoadError: MethodError: Cannot `convert` an object of type Oxygen.Core.ReviseHandler to an object of type typeof(CorsMiddleware).
```

Transferred to Issues and solved. Thanks!

**Comment by ndortega** [comment]
For those that are interested here's the related issue: 
https://github.com/OxygenFramework/Oxygen.jl/issues/231
